### PR TITLE
Support Linux callbacks

### LINKED ISSUE #124: Linux NewCallback
It would seem that NewCallback is not implemented on Linux.
Is this already in development?

### PR DESCRIPTION
**Pull request by jwijenbergh**
Fixes #124

Most of the work was already there, it's just some re-ordering. The main blocker was the ARM64 crash with callbacks which was nicely fixed by @TotallyGamerJet in https://github.com/ebitengine/purego/issues/124#issuecomment-1730540730 and merged in #165 

I had to split the callback tests into separate tests for floats as purego only supports float arguments with `CGO_ENABLED=0` currently.

Tested on my amd64 laptop, and an arm64 device. Both with `CGO_ENABLED=0` and `CGO_ENABLED=1`. The test suite passes and my GTK program leveraging https://github.com/jwijenbergh/puregotk/ also works on both architectures.

Edit: Floats are now supported with CGO_ENABLED=1 so this is basically just changing some build tags

### CODE REVIEW

**Comment by hajimehoshi** [comment]
Could you submit a PR only with `sys_unix_arm64.s` for iOS arm64 so that I can cherry-pick the change to the stable branch?

**Comment by hajimehoshi** [comment]
Or I hope @TotallyGamerJet would do that. Anway, let's wait for @TotallyGamerJet as we need their review.

Thank you for your hard work!

**Comment by jwijenbergh** [comment]
> Or I hope @TotallyGamerJet would do that. Anway, let's wait for @TotallyGamerJet as we need their review.
> 
> Thank you for your hard work!

Okay! Let's wait indeed. Makes more sense that @TotallyGamerJet makes that PR and I will then rebase this PR on top of that one once it is merged

**Comment by jwijenbergh** [comment]
Rebased.

**Comment by TotallyGamerJet** [comment]
Linux should be able to support floats even with CGO_ENABLED=1. You just need to change the build tags in syscall_cgo_linux.go to exclude amd64 and arm64 and then include them in sysycall_sysv.go, dlfcn_nocgo_linux.go and dlfn_stubs.s.

Hopefully that's all of them

File: sys_unix_amd64.s
Comment (TotallyGamerJet) [inline sys_unix_amd64.s:1]: After the CGO_ENABLED=1 fix put this file back into sys_amd64.s with an updated build tag to include linux

File: syscall_sysv.go
Comment (TotallyGamerJet) [inline syscall_sysv.go:23]: This code should stay here

File: callback_test.go
Comment (TotallyGamerJet) [inline callback_test.go:52]: When CGO_ENABLED=1 is fixed move these functions back here

**Review by TotallyGamerJet — APPROVED** [review]
LGTM!

**Comment by TotallyGamerJet** [comment]
@hajimehoshi do you want to look at it then merge?

File: callback_test.go
Comment (hajimehoshi) [inline callback_test.go:4]: I prefer alphabetical order, so let's write the arches like this:

```
//go:build darwin || (linux && (!cgo || amd64 || arm64))
```

Update the other comments too.

File: syscall_cgo_linux.go
Comment (hajimehoshi) [inline syscall_cgo_linux.go:29]: amd64/arm64

**Comment by jwijenbergh** [comment]
Thanks for the review @hajimehoshi. Should be good now

**Review by hajimehoshi — APPROVED** [review]
LGTM, thanks!

@TotallyGamerJet Please merge this!

**Comment by jwijenbergh** [comment]
Thanks!

**Pull request merged**
